lib/screens/onboarding/request_recovery.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:seeds/providers/services/http_service.dart';
+import 'package:seeds/widgets/account_name_field.dart';
+import 'package:seeds/widgets/main_button.dart';
+
+class RequestRecovery extends StatefulWidget {
+  final Function onRequestRecovery;
+  RequestRecovery({this.onRequestRecovery});
+
+  @override
+  _RequestRecoveryState createState() => _RequestRecoveryState();
+}
+
+class _RequestRecoveryState extends State<RequestRecovery> {
+  final TextEditingController accountNameController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<bool>(
+        future: checkGuardiansExists(accountNameController.text),
+        builder: (context, snapshot) {
+          return buildAccountForm(snapshot);
+        });
+  }
+
+  Widget buildAccountForm(snapshot) {
```

Comment:
I think snapshot can be null and make this crash.

Reply:
If it's null it will be unacceptable - won't crash.